### ISSUE TITLE
CLI auto-tracing es2015/commonjs paths problem

### ISSUE THREAD

**Issue by vermilion** [aurelia/cli#960]
**I'm submitting a bug report**

* **Library Version:**
1.0.0-beta.3

**Please tell us about your environment:**
* **Operating System:**
Windows 10

* **Node Version:**
8.9.4

* **NPM Version:**
5.6.0

* **Browser:**
Chrome 17

* **Language:**
TypeScript 3.1

* **Loader/bundler:**
RequireJS

**Current behavior:**

* **What is the expected behavior?**
We have to run our aurelia project on really old browser - Chrome 15-17. 
On pre 1.0 version everything worked fine

Now we get error "Use of const in strict mode".
![image](https://user-images.githubusercontent.com/2167823/47858900-17786d00-ddfe-11e8-8d40-443f3f1c5e35.png)

How to force compile _const/let_ to _var_ in cli?
How does aurelia cli auto-tracing get its dependencies? Some are from commonjs and some from es2015 folder although in package.json paths are the same
```
define('aurelia-binding',['aurelia-binding/dist/es2015/aurelia-binding'],function(m){return m;});
define('aurelia-bootstrapper',['aurelia-bootstrapper/dist/commonjs/aurelia-bootstrapper'],function(m){return m;});
define('aurelia-dependency-injection',['aurelia-dependency-injection/dist/commonjs/aurelia-dependency-injection'],function(m){return m;});
define('aurelia-framework',['aurelia-framework/dist/commonjs/aurelia-framework'],function(m){return m;});
define('aurelia-loader',['aurelia-loader/dist/commonjs/aurelia-loader'],function(m){return m;});
define('aurelia-loader-default',['aurelia-loader-default/dist/commonjs/aurelia-loader-default'],function(m){return m;});
define('aurelia-logging',['aurelia-logging/dist/commonjs/aurelia-logging'],function(m){return m;});
define('aurelia-metadata',['aurelia-metadata/dist/commonjs/aurelia-metadata'],function(m){return m;});
define('aurelia-pal',['aurelia-pal/dist/commonjs/aurelia-pal'],function(m){return m;});
define('aurelia-pal-browser',['aurelia-pal-browser/dist/commonjs/aurelia-pal-browser'],function(m){return m;});
define('aurelia-path',['aurelia-path/dist/commonjs/aurelia-path'],function(m){return m;});
define('aurelia-polyfills',['aurelia-polyfills/dist/commonjs/aurelia-polyfills'],function(m){return m;});
define('aurelia-task-queue',['aurelia-task-queue/dist/commonjs/aurelia-task-queue'],function(m){return m;});
define('aurelia-templating',['aurelia-templating/dist/es2015/aurelia-templating'],function(m){return m;});

```

* **What is the motivation / use case for changing the behavior?**
Support for old Webkit

**Comment by huochunpeng** [comment]
@EisenbergEffect @JeroenVinke this is a serious bug needs fix and release now.

As many aurelia repos have added "module" field in package.json pointing to es2015 folder. I did not foresee this issue.

Need to modify this file
https://github.com/aurelia/cli/blob/master/lib/build/amodro-trace/read/es.js

To add option to turn off strict mode
```
plugins: ['@babel/plugin-transform-modules-amd', {strictMode: false}]
```
Ref: https://babeljs.io/docs/en/babel-plugin-transform-strict-mode/

Transform-modules-amd accepts same option.

I did not research on why JavaScript does not support `const` in strict mode. You may want to investigate bit more. Thanks.

**Comment by huochunpeng** [comment]
Sorry. I am not sure does Chrome 17 support `const`?

If not, my proposed solution would not help. Any es2015 code with `const` would not work in old browsers without `const` support.

@EisenbergEffect @JeroenVinke if we want, we could bring in preset-env to the above `es.js` file to tranpile all es2015 files to support old browsers.

**Comment by huochunpeng** [comment]
@vermilion, temporary fix is to add explicit config to aurelia.json

```
{
  "name": "aurelia-binding",
  "path": "../node_modules/aurelia-binding/dist/amd"
}
```

You need few more for other aurelia packages with "module" field in their package.json. Note you can either use dist/amd or dist/commonjs.

**Comment by vermilion** [comment]
@huochunpeng no, as we can see from error message Chrome 17 doesn't support const in strict mode.. :(

**Comment by huochunpeng** [comment]
Does the temporary fix work for you?

**Comment by vermilion** [comment]
@huochunpeng yes, thank you
I had to add almost all aurelia packages from AMD folder

**Comment by jods4** [comment]
@huochunpeng Webpack transpiles nothing -- except modules of course... `import`, `export`, `require`, `define` and co. -- by itself. Transpiling JS is the job of a Webpack loader.

Webpack likes ES `import` much more than other module format. They enable quite a few additional optimizations (most notably tree shaking and scope hoisting). This is why I made `native-modules` the default distribution used by Webpack.

Basically, with webpack you should use:
- native-modules dist. if you target old browsers (i.e. old JS + new modules);
- es2015 dist. if you target new browsers (i.e. new JS + new modules).

There is no benefit to using any other distribution when using Webpack.
